File: dsAlgo_code/src/com/core/java/kg/sort/bubblesort/BubbleSort.java
package com.core.java.kg.sort.bubblesort;

public class BubbleSort {

	public static void main(String[] args) {
		
		int arr[]= {2,1,5,4,6,8,9,3,10,7,11};
		int n=arr.length-1;
		boolean isSwapContinue=true;
		int counter=1;
		int l=n;
		while(l>0 && isSwapContinue) {
			   isSwapContinue=false;
			   for(int i=0;i<=n;i++) {
				   int j=i+1;
				   if(n>j) {
					   if(arr[i]>arr[j]) {
						   int k=arr[i];
						   arr[i]=arr[j];
						   arr[j]=k;
						   isSwapContinue=true;
					   }
					 }
				}
			   l=l-1;
			   counter++;
			}
		 System.out.println("counter="+counter);
		 for(int i=0;i<=n;i++) {
			
			 System.out.println(arr[i]);
		 }
		}

	
}

File: dsAlgo_code/src/com/core/java/kg/algorithms/sort/bubblesort/BubbleSort.py
def bubbleSort(arr):
    n=len(arr);
    for i in range(n):
        swap=False;
        for j in range(n):
            if (j+1)<n:
                a=arr[j];
                if arr[j]>arr[j+1]:
                    arr[j]=arr[j+1];
                    arr[j+1]=a;
                    swap=True;
        if swap==False:
            break;


arr=[64, 34, 25, 12, 22, 11, 90]
bubbleSort(arr)
print(arr)